ral-facilities/datagateway
# Configuring links for all search results #566

## Description
This branch is based on and builds on [Make DataGateway work with API's ICAT backend](https://github.com/ral-facilities/datagateway/pull/663) and [Fixing slow search queries on bugfix/casing-json-fields-#94](https://github.com/ral-facilities/datagateway/pull/700) PRs. 

This finishes off the issue that was originally addressed by Patrick but not completed. Investigation search results in DataGateway currently provide a link to the investigation itself but datasets and datafiles do not. This fix ensures that they now do, regardless of the facility being used (ISIS, DLS or a generic link).

## Testing instructions
Use the ISIS dev URLs for testing as the code makes use of alternative casing when referring to fields. Perform some test search queries in DataGateway search (e.g. "test", "dog", etc.) and experiment with filtering by investigation, dataset and datafile. _All_ results should have a link in their first column (corresponding to their name or title) that should take the user to the appropriate place. For example, if the link is for a DLS investigation, the user should be navigated to `/browse/proposal/...` and if the link is for an ISIS investigation, the user should be navigated to `/browse/investigation/...`. Datafiles also have a column for parent dataset (with a link) and datasets have a column for parent investigation (also with a link).

- [ ] Check investigations link to investigation pages, datasets to dataset pages and datafiles to datafile pages
- [ ] Check that DLS results link to DLS pages and ISIS results link to ISIS pages
- [ ] Check the ID numbers in the constructed links tally with data received from ICAT

## Agile board tracking
Closes #566
